azure-stack-go-sdk/azure-stack-go-sdk-profiles.md

```diff
@@ -0,0 +1,302 @@
+The `Authorizer` must be set as the "authorizer" property of the resource client. There are different methods to get an Authorizer; for a complete list see [here](https://github.com/Azure/go-autorest/tree/master/autorest/adal#acquire-access-token).
+
+This section presents a common way to get authorizer tokens on Azure Stack by using `client credentials`.
+
+1. If a service principal with owner role on the subscription is available, skip this step. Otherwise create a service principal ([instructions](https://docs.microsoft.com/en-us/azure/azure-stack/azure-stack-create-service-principals)) and assign it an "owner" role scoped to your subscription ([instructions](https://docs.microsoft.com/en-us/azure/azure-stack/azure-stack-create-service-principals#assign-role-to-service-principal)). Save the service principal application ID and secret.
+2. Import `adal` package from Go-AutoRest in your code. 
+
+    ```go
+    package main
+
+    import "github.com/Azure/go-autorest/autorest/adal"
+    ```
+3. Create an OAuthConfig by using NewOAuthConfig method from `adal` module. 
+
+    ```go
+    package main
+
+    import "github.com/Azure/go-autorest/autorest/adal"
+
+    func CreateToken() (adal.OAuthTokenProvider, error) {
+        var token adal.OAuthTokenProvider
+        oauthConfig, err := adal.NewOAuthConfig("<activeDirectoryEndpoint>", "<tenantID>")
+    }
+    ```
+
+    Set `activeDirectoryEndpoint` to the value of `loginEndpoint` property that is retreived from the resource manager metadata endpoint on the previous section of this document.
+
+    Set `tenantID` value to your Azure Stack tenant ID.
+
+4. Finally, create a service principal token by using NewServicePrincipalToken method from `adal` module.
+
+    ```go
+    package main
+
+    import "github.com/Azure/go-autorest/autorest/adal"
+
+    func CreateToken() (adal.OAuthTokenProvider, error) {
+        var token adal.OAuthTokenProvider
+        oauthConfig, err := adal.NewOAuthConfig("<activeDirectoryEndpoint>", "<tenantID>")
+        token, err = adal.NewServicePrincipalToken(
+            *oauthConfig,
+            "<clientID>",
+            "<clientSecret>",
+            "<tokenAudience>")
+        return token, err
+    }
+    ```
+
+    Set `tokenAudience` to one of the values in the `audience` list that is retrieved from the resource manager metadata endpoint on the previous section of this document.
+
+    Set `clientID` to the service principal application ID saved when service principal was created on step 1.
+
+    Set `clientSecret` to the service principal application Secret saved when service principal was created on step 1.
+
+# Example
+This section shows a sample of Go code to create virtual network on Azure Stack. For complete examples of Go SDK see [Azure Go SDK samples repository](https://github.com/Azure-Samples/azure-sdk-for-go-samples). Azure stack samples are available under hybrid/ path inside service folders of the repository.
```

Comment:
Remove the example section after this paragraph. If there is a link to identical sample code then users can see that. See also the curated samples list in the ToC, which could link to Azure Stack samples.